devices/IFrameTransformToIWear/src/IFrameTransformToIWear.cpp

```diff
@@ -0,0 +1,518 @@
+    {
+
+        if (!m_frameTransformInterface->getTransform(m_targetFrame, m_rootFrame, m_matrixBuffer)) {
+            return false;
+        }
+
+        position[0] = m_matrixBuffer(0, 3);
+        position[1] = m_matrixBuffer(1, 3);
+        position[2] = m_matrixBuffer(2, 3);
+
+        for (size_t r = 0; r < 3; r++) {
+            for (size_t c = 0; c < 3; c++) {
+                m_rotationMatrixBuffer[r][c] = m_matrixBuffer(r, c);
+            }
+        }
+
+        quaternion = wearable::utils::rotationMatrixToQuaternion(m_rotationMatrixBuffer);
+        return true;
+    }
+
+private:
+    yarp::dev::IFrameTransform* m_frameTransformInterface = nullptr;
+    std::string m_rootFrame, m_targetFrame;
+    yarp::sig::Matrix m_matrixBuffer;
+    wearable::Matrix3 m_rotationMatrixBuffer;
+};
+
+class IFrameTransformToIWear::Impl
+{
+public:
+    bool firstRun = true;
+    mutable std::recursive_mutex mutex;
+
+    TimeStamp timestamp;
+    ParsedOptions options;
+
+    yarp::dev::IFrameTransform* frameTransformInterface = nullptr;
+
+    template <typename T>
+    using SensorsMap = std::map<std::string, SensorPtr<T>>;
+
+    // Sensors stored for exposing wearable::IWear
+    class PoseSensor;
+    SensorsMap<PoseSensor> poseSensorsMap;
+
+    template <typename SensorType>
+    bool isSensorAvailable(const std::string& name, const SensorsMap<SensorType>& map)
+    {
+        if (map.find(name) == map.end()) {
+            return false;
+        }
+        return true;
+    }
+};
+
+// ================================
+// WEARABLE SENSORS IMPLEMENTATIONS
+// ================================
+
+class IFrameTransformToIWear::Impl::PoseSensor : public wearable::sensor::IPoseSensor
+{
+public:
+    // ------------------------
+    // Constructor / Destructor
+    // ------------------------
+    PoseSensor(
+        IFrameTransformHandler* frameTransformHandler,
+        const wearable::sensor::SensorName aName,
+        const wearable::sensor::SensorStatus aStatus = wearable::sensor::SensorStatus::Unknown)
+        : IPoseSensor(aName, aStatus)
+        , m_frameTransformHandler(frameTransformHandler)
+    {}
+
+    ~PoseSensor() override = default;
+
+    // -------------------------
+    // IOrientationSensor interface
+    // -------------------------
+    bool getPose(Quaternion& orientation, Vector3& position) const override
+    {
+        return this->m_frameTransformHandler->getTransform(orientation, position);
+    }
+
+    // ------------------------
+    // Custom utility functions
+    // ------------------------
+    inline void setStatus(const wearable::sensor::SensorStatus aStatus) { m_status = aStatus; }
+
+private:
+    IFrameTransformHandler* m_frameTransformHandler = nullptr;
+};
+
+// TODO: implement other sensors
+
+// ======================
+// IFrameTransformToIWear
+// ======================
+
+wearable::sensor::SensorType sensorTypeFromString(std::string sensorTypeString)
```

Comment:
This method is duplicated from https://github.com/robotology/wearables/blob/master/devices/IAnalogSensorToIWear/src/IAnalogSensorToIWear.cpp#L300, better to put it in a common utilities

Reply:
Done in https://github.com/robotology/wearables/blob/master/msgs/thrift/WearableData.thrift#L173-L191